include/GoogleSync/GoogleSync.php

```diff
@@ -1143,22 +1231,36 @@ public function syncAllUsers()
     {
 
         // First we populate the array of syncable users
-        $this->setSyncUsers();
+        try {
+            $ret = $this->setSyncUsers();
+        } catch (Exception $e) {
+            $this->logger->error(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . '- setSyncUsers() Exception: ' . $e->getMessage());
+        }
+
+        if (!$ret) {
+            $this->logger->warn(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - There is no user to sync..');
+        }
 
         // We count failures here
         $failures = 0;
 
         // Then we go though the array and sync the users with doSync()
         if (isset($this->users) && !empty($this->users)) {
-            foreach ($this->users as $key => $value) {
-                $return = $this->doSync($key);
+            foreach (array_keys($this->users) as $key) {
+                $return = null;
+                try {
+                    $return = $this->doSync($key);
+                } catch (Exception $e) {
+                    $this->logger->error(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - doSync() Exception: ' . $e->getMessage());
+                }
                 if (!$return) {
-                    $GLOBALS['log']->error('Something went wrong syncing for user id: ' . $key);
+                    $this->logger->error(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - Something went wrong syncing for user id: ' . $key);
                     $failures++;
                 }
             }
         }
         if ($failures > 0) {
+            $this->logger->warn(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . $failures . ' failure(s) found at syncAllUsers method.');
             return false;
         } else {
             return true;
```

Comment:
This file contains a single class in 1266 lines, which is too large. Try to split out this class into small units or classes.

Reply:
Split into GoogleSyncBase and GoogleSync.